nautobot/extras/models/metadata.py

```diff
@@ -135,3 +138,108 @@ def save(self, *args, **kwargs):
     def delete(self, *args, **kwargs):
         # TODO: should we behave like CustomFieldChoice and raise a ProtectedError if this choice is in use anywhere?
         super().delete(*args, **kwargs)
+
```

Comment:
Do we need a custom Manager class for ObjectMetadata? It seems like at least we'd want something like `ObjectMetadata.objects.get_for_model(instance)` to be supported.